libraries/botbuilder-lg/src/activityFactory.ts

```diff
@@ -0,0 +1,361 @@
+    }
+
+    private static buildActivityFromObject(lgJObj: any): Activity {
+        let activity: any = {};
+
+        if ('type' in lgJObj && lgJObj.type === 'event') {
+            activity = this.buildEventActivity(lgJObj);
+        } else {
+            activity = this.buildMessageActivity(lgJObj);
+        }
+
+        return activity;
+    }
+
+    private static buildEventActivity(lgJObj: any): Activity {
+        let activity: any = { type: ActivityTypes.Event };
+        for (const item of Object.keys(lgJObj)) {
+            const property: string = item.trim();
+            const value: any = lgJObj[item];
+            switch (property.toLowerCase()) {
+                case '$type':
+                    break;
+                case 'name':
+                    activity.name = value.toString();
+                    break;
+                case 'value':
+                    activity.value = value.toString();
+                    break;
+                default:
+                    console.log(`Skipping unknown activity property ${ property }`);
+                    break;
+            }
+        }
+
+        return activity;
+    }
+
+    private static buildMessageActivity(lgJObj: any): Activity {
+        let activity: any = { type: ActivityTypes.Message };
+        for (const key of Object.keys(lgJObj)) {
+            const property: string = key.trim();
+            const value: any = lgJObj[key];
+
+            switch (property.toLowerCase()) {
+                case '$type':
+                    break;
+                case 'text':
+                    activity.text = value.toString();
+                    break;
+                case 'speak':
+                    activity.speak = value.toString();
+                    break;
+                case 'inputhint':
+                    activity.inputHint = value.toString();
+                    break;
+                case 'attachments':
+                    activity.attachments = this.getAttachments(value);
+                    break;
+                case 'suggestedactions':
+                    activity.suggestedActions = this.getSuggestions(value);
+                    break;
+                case 'attachmentlayout':
+                    activity.AttachmentLayout = value.toString();
+                default:
+                    console.log(`Skipping unknown activity property ${ property }`);
+                    break;
+            }
+        }
+
+        return activity;
+    }
+
+    private static getSuggestions(value: any): SuggestedActions {
+        let suggestedActions: SuggestedActions = {
+            actions : [],
+            to: []
+        };
+
+        let actions: any[] = this.normalizedToList(value);
+
+        for (const action of actions) {
+            if (typeof action === 'string') {
+                const actionStr: string = action.trim();
+                suggestedActions.actions.push({
+                    type: ActionTypes.MessageBack,
```

Comment:
>MessageBack [](start = 38, length = 11)

Unless something has changed recently... MessageBack isn't supported across by all channels. IMBack is safer.